Code/Mantid/Framework/DataHandling/src/LoadDetectorsGroupingFile.cpp

```diff
@@ -384,9 +387,12 @@ void LoadDetectorsGroupingFile::generateNoInstrumentGroupWorkspace() {
 /*
  * Initialization
  */
-LoadGroupXMLFile::LoadGroupXMLFile() {
-  mStartGroupID = 1;
-  return;
+LoadGroupXMLFile::LoadGroupXMLFile(): m_instrumentName(""),
+    m_userGiveInstrument(false), m_date(""), m_userGiveDate(false),
+    m_description(""), m_userGiveDescription(false),
+    m_pDoc(NULL), m_pRootElem(NULL),  m_groupComponentsMap(),
+    m_groupDetectorsMap(), m_groupSpectraMap(), m_startGroupID(1),
+    m_groupNamesMap() {
```

Comment:
It should only be necessary to explicitly instantiate `m_userGiveInstrument`, `m_userGiveDate`, `m_userGiveDescription`, `m_pDoc`, `m_pRootElem`, and `m_startGroupID`. The other members' default constructors do their part automatically.

The same goes for the other constructors in this pull request that have `someMap()` or `someString("")`.